### ISSUE TITLE
Regaining focus on buffer with unsaved changes raises errors

### ISSUE THREAD

**Issue by smutch** [airblade/vim-gitgutter#262]
If I regain focus on a buffer with unsaved changes then the following error message appears:

``` viml
Error detected while processing function gitgutter#all..gitgutter#process_buffer..gitgutter#
diff#run_diff:
line   65:
E94: No matching buffer for /var/folders/fg/bdcf5kqj5t1_1yx8j0fjvr4w0000gn/T/vgFjNWY/173.tex

Press ENTER or type command to continue
Error detected while processing function gitgutter#all:
line    4:
E171: Missing :endif
Press ENTER or type command to continue
```

Everything still seems to work as normal, but having to dismiss the error message all the time is quite distracting.

**Comment by airblade** [comment]
Assuming you're running the current version of vim-gitgutter, the [line generating the error](https://github.com/airblade/vim-gitgutter/blob/fe10e6474d70c382edcda84dc6bcd6fcd575a6c5/autoload/gitgutter/diff.vim#L78) calls `bwipeout {bufname}`.  According to Vim's help docs, error `E94` is thrown when a buffer whose name is a number is referenced by that name...during `bdelete`.  `bwipeout` has its own error code: E517.

I can't understand why you're seeing an E94 from `bwipeout`.

The path in your error message looks like a `tempname()` with a `.tex` extension appended, as per [here](https://github.com/airblade/vim-gitgutter/blob/fe10e6474d70c382edcda84dc6bcd6fcd575a6c5/autoload/gitgutter/diff.vim#L30).  Neither `173.tex` nor `/var/folders/fg/bdcf5kqj5t1_1yx8j0fjvr4w0000gn/T/vgFjNWY/173.tex` looks like a number.

So I can't understand why E94 would be thrown even if it could be thrown by `bwipeout`.

All in all I'm not sure what's going on.

What version of Vim are you on?  What other plugins do you use?

**Comment by smutch** [comment]
Thanks for the quick response and for taking a look at this.

I had a previously made sure I could replicate the problem when using git-gutter as the only loaded plugin.  The fact that the error still happened even when I did this made me think it was a git-gutter bug, however, your comments above convinced me that it must be something else in my vimrc...

After a bit of trial and error I narrowed the issue down to these lines in my vimrc:

``` viml
set cpoptions-=A
vmap [yank]y :w! ~/.vimbuffer<CR>
nmap [yank]y :.w! ~/.vimbuffer<CR>
set cpoptions-=a
nmap [yank]p :r ~/.vimbuffer<CR>
```

It seems the changes to `cpoptions` were the cause.  I've refactored the above in to functions to make sure that `cpoptions` is correctly returned to the original value after each read/write to `.vimbuffer` and all seems well now.

Apologies for this and thanks again for the quick response (and great plugin!).
